main.py

```diff
@@ -0,0 +1,128 @@
+import cv2
+from matplotlib import image
+import mediapipe as mp
+import numpy as np
+
+mp_drawing = mp.solutions.drawing_utils
+mp_drawing_styles = mp.solutions.drawing_styles
+mp_holistic = mp.solutions.holistic
+mp_pose = mp.solutions.pose
+cap = cv2.VideoCapture('D:\\Tenis_04_07\\Wideonly\\zverev.mp4')
+print((int(cap.get(4)), int(cap.get(3))))
+
+out = cv2.VideoWriter('D:/Tenis_04_07/Wideonly/mediapipe/test.mp4', cv2.VideoWriter_fourcc(*'MPG4'), 24, (int(cap.get(3)), int(cap.get(4))))
+TENNIS_POSE_CONNECTIONS = frozenset([(11, 12), (11, 13),
+                              (13, 15), (15, 17), (15, 19), (15, 21), (17, 19),
+                              (12, 14), (14, 16), (16, 18), (16, 20), (16, 22),
+                              (18, 20), (11, 23), (12, 24), (23, 24), (23, 25),
+                              (24, 26), (25, 27), (26, 28), (27, 29), (28, 30),
+                              (29, 31), (30, 32), (27, 31), (28, 32)])
+  
+def convert_to_gray(image):
+    image = cv2.cvtColor(image, cv2.COLOR_BGR2GRAY)
+    cv2.imshow('M ',image)
+    image = cv2.cvtColor(image, cv2.COLOR_GRAY2BGR)
+
+def mask(image):
+    mask = np.zeros(image.shape[:2], dtype="uint8")
+    kernel = np.ones((5, 5), np.uint8)
+    hsv = cv2.cvtColor(image, cv2.COLOR_BGR2HSV)
+    Lower_hsv = np.array([20, 15, 15])
+    Upper_hsv = np.array([210, 210, 210])
+    Mask = cv2.inRange(hsv, Lower_hsv, Upper_hsv)
+    Mask = cv2.erode(Mask, kernel, iterations=1)
+    Mask = cv2.morphologyEx(Mask, cv2.MORPH_OPEN, kernel)
+    Mask = cv2.dilate(Mask, kernel, iterations=1)
+    mask = cv2.bitwise_not(Mask)
+    cv2.imshow("Mask", mask)
+    masked = cv2.bitwise_and(image, image, mask=mask)
+    cv2.imshow("Mask applied to Image", masked)
+    return masked
+
+def brightness(image):
+    kernel = np.array([[0, -1, 0],
+                   [-1, 7,-1],
+                   [0, -1, 0]])
+    image= cv2.filter2D(src=image, ddepth=-10, kernel=kernel)
+    cv2.imshow('brightness',image)
+    return image
+
+def split_channels(image):
+
+    b,g,r = cv2.split(image)
+    cv2.imshow("Model Blue Image", b)
+    cv2.imshow("Model Green Image", g)
+    cv2.imshow("Model Red Image", r)
+
+    image = cv2.merge([r, g, b])
+
+    cv2.imshow('Merged', image)
+    return image
+
+
+allPoints = []
+with mp_pose.Pose(
+    min_detection_confidence=0.7,
+    min_tracking_confidence=0.8) as pose:
+  frame=0
+  
+  while cap.isOpened():
+  #while frame < 30:
+    frame = frame + 1
+    success, image = cap.read()
+    if not success:
+      print("Ignoring empty camera frame.")
+      break
+    
+    image = cv2.resize(image, (700, 450))
+
+    #image = converToGray(image)
+    #image = brightness(image)
+    image = split_channels(image)
+    #image = mask(image)
+
+    image.flags.writeable = False
+    image = cv2.cvtColor(image, cv2.COLOR_BGR2RGB)
+    results = pose.process(image)
+    results.pose_landmarks.landmark[6].x = -1
```

Comment:
This should be added as different function and described what are those 'magic' numbers such as 6,7,5,0. If you don't know - ask Mikołaj

Reply:
I. have written this in commit, this numers are points which mediapipe detect in head and we are not intrested in